File: python/easy/battleships_in_a_board.py
class Solution(object):
    def countBattleships(self, board):
        """
        :type board: List[List[str]]
        :rtype: int
        """
        if len(board) == 0:
            return 0
        ret = 0
        for i in xrange(0, len(board)):
            for j in xrange(0, len(board[i])):
                if board[i][j] == 'X':
                    if (i == 0 or board[i-1][j] == '.') and (j == 0 or board[i][j-1] == '.'):
                        ret += 1
        return ret

File: cpp/easy/battleships_in_a_board.cpp
class Solution {
public:
    int countBattleships(vector<vector<char>>& board) {
        int ret = 0; 
        if(board.size() == 0)
            return ret;
        for(int i = 0; i < board.size(); ++i)
        {
            for(int j = 0; j < board[i].size(); ++j)
            {
                if(board[i][j] == 'X')
                {
                    if((i == board.size()-1 || board[i+1][j] == '.') && (j == board[i].size()-1 || board[i][j+1] == '.'))
                        ret++;
                }
            }
        }
        return ret;
    }
};

Conclusion:
easy, no comments
